pkg/ccl/streamingccl/streamproducer/replication_stream_test.go

```diff
@@ -938,17 +938,28 @@ USE d;`)
 	codec := source.mu.codec.(*partitionStreamDecoder)
 	updateCount := 0
 	for {
-		source.mu.Lock()
-		require.True(t, source.mu.rows.Next())
-		source.mu.codec.decode()
-		if codec.e.Batch != nil {
-			for _, cfg := range codec.e.Batch.SpanConfigs {
-				if receivedSpanConfigs.maybeAddNewRecord(cfg.SpanConfig, cfg.Timestamp.WallTime) {
-					updateCount++
+		func() {
+			// This codeblock is wrapped in an anonymous function to ensure the source
+			// gets unlocked if an assertion fails. Else, the test can hang.
+			source.mu.Lock()
+			defer source.mu.Unlock()
+			require.True(t, source.mu.rows.Next())
+			source.mu.codec.decode()
```

Comment:
Not your problem, and not for your PR, but once things settle a bit we should clean up thi stet code. This source interface is a bear to use.

Reply:
Agreed.